src/feature/hs/hs_dos.c

```diff
@@ -7,14 +7,84 @@
 
 #define HS_DOS_PRIVATE
 
+#include "core/or/or.h"
+#include "app/config/config.h"
+
 #include "core/or/circuitlist.h"
 
+#include "feature/nodelist/networkstatus.h"
+#include "feature/relay/routermode.h"
+
+#include "lib/evloop/token_bucket.h"
+
 #include "hs_dos.h"
```

Comment:
I'd appreciate a file-level comment here with the aims of this subsystem and what we are hoping it achieves. I'm a bit behind on the latest experimental results with this feature, so having the goals/expectations clearly defined would be helpful to me.

Reply:
Fixup commit: 22232698ca

Reply:
LGTM!